net/prosody/Makefile

```
@@ -56,18 +53,13 @@ define Build/Configure
 		--with-lua="$(STAGING_DIR_HOSTPKG)" \
 		--with-lua-include="$(STAGING_DIR)/usr/include" \
 		--with-lua-lib="$(STAGING_DIR)/usr/lib" \
-		--cflags="$(TARGET_CFLAGS)" \
+		--cflags="$(TARGET_CFLAGS) $(FPIC) -std=gnu99" \
 		--ldflags="$(TARGET_LDFLAGS) -llua -lm -ldl -shared" \
 		--c-compiler="$(CC)" \
 		--linker="$(LD)" \
```

```suggestion
		--linker="$(CC)" \
```
This alone should fix it.  However, doing some Makefile cleanup won't hurt.
Notice that the package `configure` script uses `LD="gcc"` by default.